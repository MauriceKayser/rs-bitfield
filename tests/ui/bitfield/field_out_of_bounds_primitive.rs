//! Assertion failure: _FIELD_0_EXCEEDS_THE_BIT_FIELD_SIZE

extern crate alloc;

#[bitfield::bitfield(size)]
struct BitField(u128); // Can only store bits between 0 - (sizeof(usize) * 8).

fn main() {}