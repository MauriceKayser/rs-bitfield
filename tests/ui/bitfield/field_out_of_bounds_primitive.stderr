error[E0080]: evaluation of constant value failed
 --> $DIR/field_out_of_bounds_primitive.rs:5:1
  |
5 | #[bitfield::bitfield(size)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ attempt to compute `0_usize - 1_usize`, which would overflow
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
